src/ext/JourneyPatternStopPointMap/JourneyPatternStopPointMap.tsx

```diff
@@ -2,21 +2,57 @@ import './styles.scss';
 import FormMap from '../../components/FormMap';
 import MarkerClusterGroup from 'react-leaflet-cluster';
 import { useAppSelector } from '../../store/hooks';
-import { Reducer, useCallback, useEffect, useReducer } from 'react';
+import {
+  Reducer,
+  useCallback,
+  useEffect,
+  useMemo,
+  useReducer,
+  useRef,
+} from 'react';
 import StopPlaceMarker from './StopPlaceMarker';
 import { JourneyPatternStopPointMapProps } from './types';
-import { Polyline } from 'react-leaflet';
-import { getStopPlaces } from '../../actions/stopPlaces';
+import { Polyline, ZoomControl } from 'react-leaflet';
+import { getStopPlaces, searchStopPlaces } from '../../actions/stopPlaces';
 import { useDispatch } from 'react-redux';
-import { StopPointLocation } from '../../reducers/stopPlaces';
+import { StopPlacesState, StopPointLocation } from '../../reducers/stopPlaces';
 import { StopPlace } from '../../api';
 import QuaysWrapper from './Quay/QuaysWrapper';
+import SearchPopover from './Popovers/SearchPopover';
+import {
+  FocusedMarkerNewMapState,
+  getSelectedQuayIds,
+  onFocusedMarkerNewMapState,
+} from './helpers';
 
 interface MapState {
   quayStopPointSequenceIndexes: Record<string, number[]>;
   stopPointLocationSequence: StopPointLocation[];
   showQuaysState: Record<string, boolean>;
   hideNonSelectedQuaysState: Record<string, boolean>;
+  focusedMarker: FocusedMarker | undefined;
+}
+
+/**
+ * Determines whether to "focus" a stop place or a quay
+ */
+export enum JourneyPatternMarkerType {
+  QUAY = 'quay',
+  STOP_PLACE = 'stop_place',
+}
+
+export interface JourneyPatternMarker {
+  id: string;
+  type: JourneyPatternMarkerType;
+}
+
+/**
+ * E.g. when user clicks on "locate" button in the search input results,
+ * a certain marker gets into a "focused" state - meaning e.g. its popup opens up
+ */
+export interface FocusedMarker {
+  stopPlace: StopPlace;
+  marker: JourneyPatternMarker;
 }
 
 export const JourneyPatternStopPointMap = ({
```

Comment:
this component is getting quite crowded with hooks - which probably contributes to my difficulty following the changes. Could be worth extracting hooks / business logic and document them better.

Reply:
This got moved away to types.ts